controllers/ssp_controller.go

```diff
@@ -186,7 +198,95 @@ func cleanup(request *common.Request) error {
 	return err
 }
 
+func patchCRs(sspRequest *common.Request, kinds []string, patch []byte) error {
+	for _, kind := range kinds {
+		crs := &unstructured.UnstructuredList{}
+		crs.SetKind(kind)
+		crs.SetAPIVersion("ssp.kubevirt.io/v1")
+		err := sspRequest.Client.List(sspRequest.Context, crs)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				sspRequest.Logger.Info(fmt.Sprintf("No %s CR found", kind))
+				continue
+			} else {
+				sspRequest.Logger.Error(err, fmt.Sprintf("Error listing %s CRs: %s", kind, err))
+				return err
+			}
+		}
+		for _, item := range crs.Items {
+			err = sspRequest.Client.Patch(sspRequest.Context, &item, client.RawPatch(types.MergePatchType, patch))
+			if err != nil {
+				// Patching failed, maybe the CR just got removed? Log an error but keep going.
+				sspRequest.Logger.Error(err, fmt.Sprintf("Error pausing %s from namespace %s: %s",
+					item.GetName(), item.GetNamespace(), err))
+			}
+		}
+	}
+
+	return nil
+}
+
+func preReconcile(sspRequest *common.Request, kinds []string, operatorVersion string) error {
+	preReconcilePatch := []byte(`{
+  "metadata":{
+    "annotations":{"kubevirt.io/operator.paused": "true"}
+  },
+  "status":{
```

Comment:
The `*version` status fields should only be updated on the new `SSP` CR, the old ones are irrelevant and should only be paused and eventually removed by HCO.
`sspRequest.instance.Status.*Version` is where the version values should be

Reply:
The version fields looks like a separate functionality, maybe it can be a new PR?

Reply:
My wording to describe this task probably wasn't the best, but I would like a single PR to include the entire upgrade path of the operator, and the `version` fields are part of it